### PR TITLE
statistics: test `tidb_partition_prune_mode` session variable

### PR DESCRIPTION
**Pull request by rebelice**
Problem Summary:

test `tidb_partition_prune_mode` session variable

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: statistics/handle/handle_test.go [hunk 1]
```diff
@@ -1512,6 +1512,49 @@ func (s *testStatsSuite) TestAnalyzeWithDynamicPartitionPruneMode(c *C) {
...
+		"TableReader 5.00 root partition:all data:TableFullScan",
+		"└─TableFullScan 5.00 cop[tikv] table:t keep order:false",
+	))
+	tk2.MustQuery("explain format = 'brief' select * from t").Check(testkit.Rows(
+		"PartitionUnion 5.00 root  ",
+		"├─TableReader 3.00 root  data:TableFullScan",
+		"│ └─TableFullScan 3.00 cop[tikv] table:t, partition:p0 keep order:false",
+		"└─TableReader 2.00 root  data:TableFullScan",
+		"  └─TableFullScan 2.00 cop[tikv] table:t, partition:p1 keep order:false",
+	))
```
Comment (qw4990) [inline statistics/handle/handle_test.go:1555]: Please then enable prune-mode on tk2 and disable it on tk1 and then check if it works right again.

**Review by Reminiscent — COMMENTED** [review]
Please fix the test, resolve the [comment](https://github.com/pingcap/tidb/pull/23176/files#r589363499) and conflicts.

**Review by Reminiscent — APPROVED** [review]
/LGTM

**Pull request merged**